error.go

```diff
@@ -40,7 +40,7 @@ type Errors []error
 // Error concatenates the list of error strings to provide a single string
 // that can be used to represent the errors that occurred.
 func (e Errors) Error() string {
-	var errors []string
+	errors := make([]string, len(e))
```

Comment:
A more idiomatic and concise way:

```
errors := append([]string{}, e...) // no for loop needed!
```

That's actually optimized by the compiler.

Also, don't forget about `strings.Builder`: https://pkg.go.dev/strings#Builder
It can be more verbose, but it's *way* more efficient.  Depending on how often this method is called, the following can result in big performance wins:

```
var output strings.Builder
output.Write([]byte("multiple errors: "))
for i, msg := range e {
    if i > 0 {
        output.WriteRune(',')
    }

    output.WriteString(msg)
}

```

You can also do the above with a good ol' `bytes.Buffer`, but `strings.Builder` is heavily optimized around string copying and avoiding memory allocations.